ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -472,10 +526,73 @@ void record_endpoint(VALUE recorder_instance, uint64_t local_root_span_id, ddog_
   }
 }
 
+#define MAX_LEN_HEAP_ITERATION_ERROR_MSG 256
+
+// Heap recorder iteration context allows us access to stack recorder state and profile being serialized
+// during iteration of heap recorder live objects.
+typedef struct heap_recorder_iteration_context {
+  struct stack_recorder_state *state;
+  ddog_prof_Profile *profile;
+  bool error;
+  char errorMsg[MAX_LEN_HEAP_ITERATION_ERROR_MSG];
```

Comment:
Minor: Suggest using snake case ;)

```suggestion
  char error_msg[MAX_LEN_HEAP_ITERATION_ERROR_MSG];
```